syntax = "proto3";
import "map_renderer.proto";
import "transport_router.proto";
package protobuf;
 
message Stop {
	uint32 id = 1;
    string name = 2;
    double latitude = 3;
    double longitude = 4;
}
 
message Bus {
    string name = 1;
    repeated uint32 stops = 2;
    bool is_circle = 3;  
}
 
message Distance {
    uint32 start = 1;
    uint32 end = 2;
    uint32 distance = 3;
}
 
message TransportCatalogue {
    repeated Stop stops = 1;
    repeated Bus buses = 2;
    repeated Distance distances = 3;
    RenderSettings render_settings = 4;
    RoutingSettings routing_settings = 5;
}

